include/pms-utils/misc/try_parse.hpp

```diff
@@ -0,0 +1,21 @@
+#include "boost/parser/parser.hpp"
+
+#include <expected>
+#include <iterator>
+#include <string_view>
+
+namespace pms_utils::misc {
+template <typename Rule, typename It>
```

Comment:
`It` is a misnomer since this isn't an iterator, it's an iterable
perhaps use `std::ranges::range Range` ?